### DISCUSSION TITLE
FVM (Experimental) Mainnet v15 Testing & Feedback

### DISCUSSION THREAD

**Discussion by jennijuju** [filecoin-project/lotus#8334] [Build and Sync]
🍼 Filecoin v16 Skyr upgrade will introduce Filecoin non-programmable WASM-based [Filecoin Virtual Machine (FVM)](https://fvm.filecoin.io) to Filecoin mainnet. Upon the v16 Skyr upgrade, Mainnet will atomically switch from the current built-in VMs to th[e WASM-based reference FVM](https://github.com/filecoin-project/ref-fvm). A new gas model will be activated that accounts for actual WASM execution costs. Only [built-in actors](https://github.com/filecoin-project/builtin-actors) will be supported at this time. This upgrade will mark the completeness of FVM development Milestone 1.

That being said, before fully switch the network using FVM and built-in actor, we'd like to make sure it works with the existing the network. Meaning, for any implementation that iterates the FVM and built-in actor, the node should be able to syncing the current network, SPs may producing blocks, proving the storage, sealing new sectors and making deals, and the rest of the network operation, including msig, payment channel, verifreg and so on should remain functional.

## Regular Operations
To start with, update your nodes to lotus [v1.15.1-rc2](https://github.com/filecoin-project/lotus/releases/tag/v1.15.1-rc2), set the `LOTUS_USE_FVM_EXPERIMENTAL=1` and restart your node.

Note:  the current lotus build can only sync mainnet using FVM!

Then make a copy of the [FVM 0.5 Checklist](https://github.com/filecoin-project/lotus/discussions/8334#discussioncomment-2433491),  ✅ boxes off as you perform the operation more than once and leave any comments/feedback in the call out session!

## **⚠️ Reminders and Risk Mitigations**
- **Questions are welcomed in this discussion, the team will be monitoring this discussion and check-in regularly.**
- Issues can be reported in the https://github.com/filecoin-project/lotus/issues. **Issues should be submitted with:**
    - daemon logs 
    - go routine dumps ** lotus pprof goroutines**
    - the operation/step at which the problem occurred
    - The stack trace or logs as relevant. As always, for large log files, you may upload it in this [gdrive folder](https://drive.google.com/drive/folders/1PhsWKvzrl8-Zn8ZExvzahsKoGrSaApy3) and link it in the issue!
- If you think FVM is not working well, keep all the logs for describing the issue, then disable `LOTUS_USE_FVM_EXPERIMENTAL` and restart your node!

**Comment by jennijuju** [comment]
## Ozzy's FVM 0.5 Checklist

(order doesn't matter)
- [ ] Node stays in sync.
- [ ] Pledge a new sector.
- [ ] Extend a sector expiration. 
- [ ] Make a snap deal.
- [ ] Add a new control address 
- [ ] Make a paid retrieval deal (self dealing if there is no retrieval client!). 
- [ ] Make a WindowPoSt.
- [ ] Get luck and mine a new block!
- [ ] Collect the following metrics
- [ ] Disable FVM and restart the node, operations remain functional.


|    | FVM  | Lotus Legacy VM   | Perf +- %  |
|---|---|---|---|
| block validation time  |   |   |   |
| memory usage  |   |   |   |
| CPU usage  |   |   |   |

### Any callouts

- I'm excited for FVM!
![image](https://user-images.githubusercontent.com/42981373/160020976-bc688ebb-17e3-48ea-a0f3-a84483af1bea.png)

**Comment by TippyFlitsUK** [comment]
## TippyFlits FVM 0.5 Checklist

- [x] Node stays in sync.
- [x] Pledge a new sector.
- [x] Extend a sector expiration. 
- [x] Make a snap deal.
- [x] Add a new control address 
- [x] Make a paid retrieval deal (self dealing if there is no retrieval client!). 
- [x] Make a WindowPoSt.
- [x] Get luck and mine a new block!
- [x] Collect the following metrics
- [x] Disable FVM and restart the node, operations remain functional.


|    | FVM  | Lotus Legacy VM   | Perf +- %  |
|---|---|---|---|
| block validation time  |   |   |  Unchanged |
| memory usage  |   |   | Unchanged  |
| CPU usage  |   |   | Unchanged  |

**Reply by jennijuju** [reply]
@TippyFlitsUK we may have to uncheck 1 box lolol

**Reply by TippyFlitsUK** [reply]
..... so close to a full house!!

**Comment by Angelo-gh3990** [comment]
## Angelo's FVM 0.5 Checklist
* [x]  Node stays in sync.
* [x]  Pledge a new sector.
* [x]  Extend a sector expiration.
* [x]  Make a snap deal.
* [ ]  Add a new control address
* [x]  Make a paid retrieval deal (self dealing if there is no retrieval client!).
* [x]  Make a WindowPoSt.
* [ ]  Get luck and mine a new block!
* [ ]  Collect the following metrics
* [ ]  Disable FVM and restart the node, operations remain functional.

|  | FVM | Lotus Legacy VM | Perf +- % |
| ------------- | ------------- | ------------- | ------------- |
| block validation time  |   |   |   |
| memory usage  |   |   |   |
| CPU usage  |   |   |   |

Updates will follow!

**Comment by rjan90** [comment]
Phi-rjan´s FVM 0.5 Checklist
---

- [x] Node stays in sync.
- [x] Pledge a new sector.
- [x] Extend a sector expiration.
- [x] Make a snap deal.
- [x] Add a new control address
- [x] Make a paid retrieval deal (self dealing if there is no retrieval client!).
- [x] Make a WindowPoSt.
- [x] Get lucky and mine a new block!
- [x] Disable FVM and restart the node, operations remain functional.
- [x] Collect the following metrics

|  | FVM | Lotus Legacy VM | Perf +- % 
| -- | -- | -- | --
block validation time | 0.498s  | 0.503s  |  -
memory usage | 16GiB  |16GiB|  -
CPU usage | -  | -  | unchanged

**Comment by jamerduhgamer** [comment]
## James' FVM 0.5 Checklist

- [x] Node stays in sync.
- [x] Pledge a new sector.
- [x] Extend a sector expiration. 
- [ ] Make a snap deal.
- [x] Add a new control address 
- [ ] Make a paid retrieval deal (self dealing if there is no retrieval client!). 
- [x] Make a WindowPoSt.
- [x] Get luck and mine a new block!
- [x] Disable FVM and restart the node, operations remain functional.
- [x] Collect the following metrics


|    | FVM  | Lotus Legacy VM   | Perf +- %  |
|---|---|---|---|
| block validation time  | <1 second | <1 second | 0% |
| memory usage  | ~14 GiB | ~14 GiB | 0% |
| CPU usage  | ~3 threads  | ~3 threads | 0% |

**Reply by jamerduhgamer** [reply]
Ran into an issue with sectors expiring immediately upon pledging a new sector. 
![image](https://user-images.githubusercontent.com/16791309/161456933-74645ccd-b981-4b83-8811-5f0b8ca948d6.png)

**Reply by jamerduhgamer** [reply]
Figured out the issue with the sectors expiring immediately.

**Comment by jennijuju** [comment]
[v1.15.1-rc3](https://github.com/filecoin-project/lotus/releases/tag/v1.15.1-rc3) is out with the actor bug fix!

**Comment by Reiers** [comment]
Reiers´s FVM 0.5 Checklist
---

- [x] Node stays in sync.
- [x] Pledge a new sector.
- [x] Extend a sector expiration.
- [x] Make a snap deal.
- [x] Add a new control address
- [ ] Make a paid retrieval deal (self dealing if there is no retrieval client!).
- [x] Make a WindowPoSt.
- [x] Get lucky and mine a new block!
- [x] Disable FVM and restart the node, operations remain functional.
- [ ] Collect the following metrics

|  | FVM | Lotus Legacy VM | Perf +- % 
| -- | -- | -- | --
block validation time |   | TBD  |  
memory usage |   | TBD |  
CPU usage |   |   |  

Will have production upgraded by the end of the day.

**Comment by Reiers** [comment]
[v1.15.1-rc4](https://github.com/filecoin-project/lotus/releases/tag/v1.15.1-rc4) is out.

**Comment by cryptowhizzard** [comment]
## Wijnand's FVM Checklist. ( Running v1.15.1-rc4 )

- [X]  Node stays in sync.
- [X]  Pledge a new sector.
- [ ] Extend a sector expiration. 
- [ ] Make a snap deal.
- [ ] Add a new control address - [X] 
- [X] Make a paid retrieval deal (self dealing if there is no retrieval client!). 
- [X ] Make a WindowPoSt.
- [X] Get luck and mine a new block!
- [ ] Collect the following metrics
- [ ] Disable FVM and restart the node, operations remain functional.

|    | FVM  | Lotus Legacy VM   | Perf +- %  |
|---|---|---|---|
| block validation time  |  No Change 
| memory usage  |   No Change 
| CPU usage  |  No Change 

More updates coming ;-)
